lib/blobStorage.ts

```diff
@@ -563,26 +632,58 @@ export class AzureBlobStorage extends BlobStorage {
       .then(readEntireString);
   }
 
-  public setBlobContentsFromString(blobPath: string | BlobPath, blobContents: string): Promise<void> {
+  public setBlobContentsFromString(blobPath: string | BlobPath, blobContents: string, options?: BlobContentOptions): Promise<void> {
     return this.getBlockBlobURL(blobPath)
-      .upload(azure.Aborter.none, blobContents, Buffer.byteLength(blobContents, "utf-8"))
+      .upload(azure.Aborter.none, blobContents, Buffer.byteLength(blobContents, "utf-8"), {
+        blobHTTPHeaders: {
+          blobContentType: options && options.contentType
+        }
+      })
       .then(() => { });
   }
 
+  public getBlobContentType(blobPath: string | BlobPath): Promise<string | undefined> {
+    blobPath = BlobPath.parse(blobPath);
+    const containerName: string = blobPath.containerName;
+    return this.getBlockBlobURL(blobPath)
+      .getProperties(azure.Aborter.none)
+      .then((properties: azure.Models.BlobGetPropertiesResponse) => properties.contentType)
+      .catch((error: Error) => {
+        return (error as any).statusCode !== 404
+          ? Promise.reject(error)
+          : this.containerExists(containerName)
+              .then((exists: boolean) =>
+                Promise.reject(new Error(!exists
+                  ? "ContainerNotFound: The specified container does not exist."
+                  : "BlobNotFound: The specified blob does not exist.")));
+      });
+  }
+
+  public setBlobContentType(blobPath: string | BlobPath, contentType: string): Promise<void> {
+    return this.getBlockBlobURL(blobPath)
+      .setHTTPHeaders(azure.Aborter.none, {
+        blobContentType: contentType
+      })
+      .then(() => {});
```

Comment:
Is that then part necessary?

Reply:
That's what converts it to a Promise<void>.